pkg/scheduler/core/generic_scheduler.go

```diff
@@ -452,6 +464,65 @@ func (g *genericScheduler) ScaleSchedule(ctx context.Context, placement *policyv
 	return result, nil
 }
 
+func (g *genericScheduler) scaleScheduleWithReplicaDivisionPreferenceAggregated(spec *workv1alpha1.ResourceBindingSpec) (result ScheduleResult, err error) {
+	assignedReplicas := util.GetSumOfReplicas(spec.Clusters)
+	if assignedReplicas > spec.Resource.Replicas {
+		newTargetClusters, err := g.scaleDownScheduleWithReplicaDivisionPreferenceAggregated(spec)
+		if err != nil {
+			return result, fmt.Errorf("failed to scaleDown: %v", err)
+		}
+		result.SuggestedClusters = newTargetClusters
+	} else if assignedReplicas < spec.Resource.Replicas {
+		newTargetClusters, err := g.scaleUpScheduleWithReplicaDivisionPreferenceAggregated(spec)
+		if err != nil {
+			return result, fmt.Errorf("failed to scaleUp: %v", err)
+		}
+		result.SuggestedClusters = newTargetClusters
+	} else {
+		result.SuggestedClusters = spec.Clusters
+	}
+	return result, nil
+}
+
+func (g *genericScheduler) scaleDownScheduleWithReplicaDivisionPreferenceAggregated(spec *workv1alpha1.ResourceBindingSpec) ([]workv1alpha1.TargetCluster, error) {
+	return g.divideReplicasAggregatedWithClusterReplicas(spec.Clusters, spec.Resource.Replicas)
+}
+
+func (g *genericScheduler) scaleUpScheduleWithReplicaDivisionPreferenceAggregated(spec *workv1alpha1.ResourceBindingSpec) ([]workv1alpha1.TargetCluster, error) {
+	// find the clusters that have old replicas so we can assign new replicas to them preferentially
+	targetMap := make(map[string]int32)
```

Comment:
What's the `targetMap`? The final answer(final selected cluster and replicas assigned to it)?

Reply:
It is a map of the result  for the old replicas. So that we can merge it with the new result easily